source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/admin/settings/stores/-components/form.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useMemo } from 'react';
+import { withValue } from 'react-state-helpers';
+import { attributesFor, useOrbit, idFromRecordIdentity, query } from 'react-orbitjs';
+import { Dropdown } from 'semantic-ui-react';
+
+import { buildOptions } from '~/data';
+
+import { isEmpty, hasValues } from '@lib/collection';
+import { useTranslations } from '@lib/i18n';
+import * as toast from '@lib/toast';
+
+function getStoreType(dataStore, store) {
+  if (!store) return;
+
+  return dataStore.cache.query((q) => q.findRelatedRecord(store, 'storeType'));
+}
+
+function StoreForm({ save, cancel, store, storeTypes }) {
+  const { t } = useTranslations();
+  const { dataStore } = useOrbit();
+
+  const attributes = attributesFor(store || {});
+  const isEditing = !!store;
+
+  const existingStoreType = useMemo(() => getStoreType(dataStore, store), [(store || {}).id]);
+
+  // Form States
+  const [name, setName] = useState(attributes.name || '');
+  const [storeType, setStoreType] = useState(existingStoreType);
+  const [description, setDescription] = useState(attributes.description || '');
+  const [errors, setError] = useState<IMap<string>>({});
+  // End Form States
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    let newErrors = {
+      name: isEmpty(name) && t('errors.requiredField', { field: t('stores.attributes.name') }),
+    };
+
+    setError(newErrors);
+    const isValid = !hasValues(Object.values(newErrors));
+
+    if (isValid) {
+      try {
+        await save({ name, description }, { storeType });
+      } catch (e) {
+        toast.error(e);
+      }
+    }
+  };
+
+  return (
+    <>
```

Comment:
I can't say I have ever seen this. Intentional?

Reply:
it is, it's how you tell react that you're returning multiple top-level children elements. :)